### ISSUE TITLE
Multi-user support with auth and session cookies

### ISSUE THREAD

**Issue by nfagerlund** [nfagerlund/eardogger#4]
def file under "minimum viable."

- looks like passport.js is pretty promising. has many auth methods. It'd be cool to support github and DW's openid or something, as well as just username and password. 
- For passwords, gotta store in DB with bcrypt. do I need to learn what a "salt" is?? 
- Do I want to support password resets, or just say it's your problem and I don't want your email address? 
    - THE LATTER.
- Should I start with just github, for ease of use / minimum-viability? probs

**Comment by nfagerlund** [comment]
OK, I got the tables set up right and I got all the dogear model methods working correctly with the passed userID. But:

- [ ] The app code still hardcodes userID to 1. 
- [ ] No model methods for working with users.

**Comment by nfagerlund** [comment]
Wow, I guess a lot of other stuff also happened since I wrote that ticket description, huh. 

- I went with passport. 
- I went with the local username/password authentication strat. 
- I'll probably have to add API tokens once I start working on an extension, and I might later decide to add "log in with," but probably not. 
- MVP will be manual user management where I send you your password on a secure channel, sign-up and account edit come later.
